RecoCTPPS/PixelLocal/plugins/CTPPSPixelLocalTrackProducer.cc

```diff
@@ -0,0 +1,258 @@
+
+#include "RecoCTPPS/PixelLocal/interface/CTPPSPixelLocalTrackProducer.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+
+#include "RecoCTPPS/PixelLocal/interface/RPixRoadFinder.h"
+#include "RecoCTPPS/PixelLocal/interface/RPixPlaneCombinatoryTracking.h"
+
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+
+#include "CLHEP/Vector/ThreeVector.h"
+#include "CLHEP/Vector/RotationInterfaces.h"
+#include "DetectorDescription/Core/interface/DDRotationMatrix.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+//------------------------------------------------------------------------------------------------//
+
+CTPPSPixelLocalTrackProducer::CTPPSPixelLocalTrackProducer(const edm::ParameterSet& parameterSet) :
+  parameterSet_(parameterSet){
+
+  inputTag_             = parameterSet_.getParameter<std::string>  ("label"                  );
+  verbosity_            = parameterSet_.getUntrackedParameter<int> ("RPixVerbosity"          );
+  maxHitPerRomanPot_    = parameterSet_.getUntrackedParameter<int> ("RPixMaxHitPerRomanPot"  );
+  maxHitPerPlane_       = parameterSet_.getUntrackedParameter<int> ("RPixMaxHitPerPlane"     );
+  maxTrackPerRomanPot_  = parameterSet_.getUntrackedParameter<int> ("RPixMaxTrackPerRomanPot");
+  maxTrackPerPattern_   = parameterSet_.getUntrackedParameter<int> ("RPixMaxTrackPerPattern" );
+  numberOfPlanesPerPot_ = parameterSet_.getUntrackedParameter<int> ("NumberOfPlanesPerPot"   );
```

Comment:
all parameters that can change results should be tracked (exceptions are for printouts and similar diagnostic or decision to produce or not some products)